cli/azd/pkg/commands/pipeline/pipeline_manager.go

```diff
@@ -198,15 +203,14 @@ func validateDependencyInjection(ctx context.Context, manager *PipelineManager)
 	if manager.CiProvider == nil {
 		log.Panic("missing CI provider for pipeline manager")
 	}
-	if input.GetConsole(ctx) == nil {
-		log.Panic("missing input console in the provided context")
+	if manager.console == nil {
```

Comment:
nit. Let's centralize all of these input validations to the `New<ComponentName>()` functions.

In general, if there's an interface as a dependency for creating an object, the object should expose a constructor to validate that the interface is not null. 

Does it make sense?

Reply:
Makes sense - can do.